src/main/java/com/rtb/projectmanagementtool/task/TaskController.java

```diff
@@ -42,6 +43,77 @@ private void addKeysToTasks(ArrayList<TaskData> tasks, ArrayList<Key> keys) {
     }
   }
 
+  // Update methods
+
+  public void addUser(long taskID, long userID) {
+    addUser(getTaskByID(taskID), userID);
+  }
+
+  public void addUser(TaskData task, long userID) {
+    if (task.getTaskID() != 0) {
+      TransactionOptions options = TransactionOptions.Builder.withXG(true);
+      Transaction transaction = datastore.beginTransaction(options);
+      try {
+        addUserRecursive(task, userID);
+        transaction.commit();
+      } finally {
+        if (transaction.isActive()) {
+          transaction.rollback();
+        }
+      }
+    }
+  }
+
+  private void addUserRecursive(TaskData task, long userID) {
```

Comment:
Maybe ok for now, but for bigger projects definitely avoid doing recursive database operations.  Transactions does protect us from failures in between.  But we'd still be doing multiple back-and-forths with the database server which may take some time.